config/v1/types_infrastructure.go

```diff
@@ -123,6 +123,60 @@ const (
 	ExternalTopologyMode TopologyMode = "External"
 )
 
+// PlatformLoadBalancerType defines the type of load balancer used by the cluster.
+type PlatformLoadBalancerType string
+
+const (
+	// LoadBalancerTypeUserManaged is a load balancer with control-plane VIPs managed outside of the cluster by the customer.
+	LoadBalancerTypeUserManaged PlatformLoadBalancerType = "UserManaged"
+
+	// LoadBalancerTypeOpenShiftManaged is a load balancer with control-plane VIPs managed by the OpenShift cluster.
+	LoadBalancerTypeOpenShiftManaged PlatformLoadBalancerType = "OpenShiftManaged"
+)
+
+// PlatformLoadBalancerOpenShiftManaged defines the type of the OpenShift managed load balancer used by the cluster.
+type PlatformLoadBalancerOpenShiftManagedType string
+
+const (
+	// OpenShiftManagedTypeDefault is a load balancer managed by OpenShift with default configuration (currently Keepalived + HAProxy).
+	OpenShiftManagedTypeDefault PlatformLoadBalancerOpenShiftManagedType = "Default"
+)
+
+// PlatformLoadBalancerOpenShiftManaged defines the type of the OpenShift managed load balancer used by the cluster.
+type PlatformLoadBalancerOpenShiftManaged struct {
+	// Type is the type of the OpenShift managed load balancer used by the cluster.
+	// The default value is Default.
+	// +default="Default"
+	// +kubebuilder:default:="Default"
+	// +kubebuilder:validation:Enum="Default"
+	// +kubebuilder:validation:Required
+	// +optional
+	Type PlatformLoadBalancerOpenShiftManagedType `json:"type,omitempty"`
+}
+
+// BareMetalPlatformLoadBalancer defines the load balancer used by the cluster on BareMetal platform.
+type BareMetalPlatformLoadBalancer struct {
```

Comment:
Please check the api-conventions docs (enhancements/dev-guide) for the guidance around discriminated unions, there's a few tags you need to add and the members (OpenShiftManaged) should be pointers (one very rare case of a pointer)